File: day_6/6.js
/** @param {string[]} gifts
 *  @returns {boolean} True if the gift is inside the box
 */
function inBox(box) {
  return (
    box.every((row) => row[0] === "#" && row[row.length - 1] === "#") &&
    box.slice(1, -1).some((row) => row.includes("*"))
  );
}

console.log(inBox(["###", "#*#", "###"])); // ➞ true
console.log(inBox(["####", "#* #", "#  #", "####"])); // ➞ true
console.log(inBox(["#* #", "#  #", "####"])); // ➞ false
console.log(inBox(["*####", "#   #", "#  #*", "####"])); // ➞ false
console.log(inBox(["#####", "#   #", "#   #", "#   #", "#####"])); // ➞ false

File: day_6/6.py
def in_box(box):
    return all(row[0] == "#" and row[-1] == "#" for row in box) and any(
        "*" in row for row in box[1:-1]
    )


print(in_box(["###", "#*#", "###"]))  # ➞ true
print(in_box(["####", "#* #", "#  #", "####"]))  # ➞ true
print(in_box(["#* #", "#  #", "####"]))  # ➞ false
print(in_box(["*####", "#   #", "#  #*", "####"]))  # ➞ false
print(in_box(["#####", "#   #", "#   #", "#   #", "#####"]))  # ➞ false
